api/types.go

```diff
@@ -750,6 +801,26 @@ func (po *PinOptions) FromQuery(q url.Values) {
 	if rpl, err := strconv.Atoi(rplStrMax); err == nil {
 		po.ReplicationFactorMax = rpl
 	}
+
+	if shsize, err := strconv.ParseUint(q.Get("shard-size"), 10, 64); err == nil {
+		po.ShardSize = shsize
+	}
+
+	if allocs := q.Get("user-allocations"); allocs != "" {
+		po.UserAllocations = strings.Split(allocs, ",")
+	}
+
+	po.Metadata = make(map[string]string)
+	for k := range q {
+		if !strings.HasPrefix(k, pinOptionsMetaPrefix) {
```

Comment:
What else are we potentially storing in the `metadata` map?

Reply:
nothing. But the query arguments that go to the `metadata` map start with `meta-`. So if there is a query argument that isn't like that I don't want to put it as metadata.